dicoogle/src/main/java/pt/ua/dicoogle/plugins/PluginController.java

```diff
@@ -603,8 +622,35 @@ public JointQueryTask query(JointQueryTask holder, List<String> querySources, fi
         });
         //logger.info("Prepared Query Task: QueryString");
         return queryTask;
-    }        
- 
+    }
+
+
+    private Task<Iterable<SearchResult>> getTaskForQueryDim(final String querySource, final String query,
+                                                         final DimLevel level, final Object ... parameters){
+
+        final QueryInterface queryEngine = getQueryProviderByName(querySource, true);
+        //returns a tasks that runs the query from the selected query engine
+        String uid = UUID.randomUUID().toString();
+        Task<Iterable<SearchResult>> queryTask = new Task<>(uid, querySource,
+                new Callable<Iterable<SearchResult>>(){
+                    @Override public Iterable<SearchResult> call() throws Exception {
+                        if(queryEngine == null || !(queryEngine instanceof QueryDimInterface)) return Collections.emptyList();
+                        try {
+                            return queryEngine.query(query, level, parameters);
+                        } catch (RuntimeException ex) {
+                            logger.warn("Query plugin {} failed unexpectedly", querySource, ex);

```

Comment:
A previously unattended issue: should public `query` methods in `DicooglePlatform` swallow the query-related exceptions `QueryException` and `RetrievalException`? Can we make them appear when attempting to get the outcome of the task?

Reply:
LGTM!